src/Lecture2.hs

```diff
@@ -158,16 +178,85 @@ You're free to define any helper functions.
 -}
 
 -- some help in the beginning ;)
+data Chest = Chest
+    { chestGold     :: Int
+    , chestTreasure :: Int
+    }
+    deriving (Show)
+
+add :: Chest -> Chest -> Chest
+add c1 c2 = Chest (chestGold c1 + chestGold c2)
+                  (chestTreasure c1 + chestTreasure c2)
```

Comment:
So... for this one, I really wanted to "overload +" (not sure if that's the right term in Haskell).
However, once I started, at any point I used `+` in the code, the compiler complained that it didn't know whether it should use the `+` that I defined or `Prelude.+` --- at which point I backed off and decided to use `add`.

What would have been the way forward if I had insisted defining `(+)`?
Making `Chest` an instance of `Num` seemed overkill...

Reply:
With the `chestTreasure :: Int` definition, you can indeed add both chests. This task was a bit vague in terms of what the treasure could mean, but in the ideal scenario I was thinking about something like the following (as in the lecture slides):

```haskell
data Chest a = Chest
    { chestGold     :: Int
    , chestTreasure :: a
    }
```

Not sure what `chestTreasure :: Int` could represent 🤔 Maybe the number of gems 😅 💎 
But it doesn't really matter. It would be enough to just return `Knight` and the result `Chest` in the `fight` function outcome and leave dealing with to the consumer of the function (so you don't need `add` and `updateFighters` in this design).

<hr>

In any case, if you want to add `Chest` then yes, implementing `Num` is a bit of an overkill indeed. In Lecture 3 I talk about the Semigroup typeclass, and you can use it instead.